src/Resources/public/core.js

```diff
@@ -941,20 +941,24 @@ var Backend =
 				alert('Could not find the SimpleModal frame');
 				return;
 			}
-			ul = frm.document.getElementById(opt.id);
+			ul = frm.document.getElementById('tl_select');
 			inp = ul.getElementsByTagName('input');
 			for (i=0; i<inp.length; i++) {
 				if (inp[i].checked && !inp[i].id.match(/^(check_all_|reset_)/)) {
 					val.push(inp[i].get('value'));
 				}
 			}
-			if (opt.tag && (field = $(opt.tag))) {
+			if (opt.callback) {
+				opt.callback(ul.get('data-table'), val);
+			} else if (opt.tag && (field = $(opt.tag))) {
+				console.warn('Using the modal selector without a callback function is deprecated. Please adjust your Contao DCA file.');
```

Comment:
`console.warn()` is not available in all browsers I think (e.g. IE9). We should wrap a `if (window.console && console.warn) {` around it to prevent errors.